content/en/docs/tutorials/services/pods-and-endpoint-termination-flow.md

```diff
@@ -0,0 +1,202 @@
+---
+title: Pods And Endpoints Termination Flow
+content_type: tutorial
+weight: 60
+---
+
+
+<!-- overview -->
+
+## The Kubernetes model for connecting containers
+
+Now that you have a continuously running, replicated application, that is exposed on a network, let's look at Pods termination flow and ways to implement a graceful connections draining.
+
+There are often a need to terminate a Pod - be it for upgrade or scale down. In order to improve application availability, it may be important to implement a proper active connections draining. This tutorial explains the flow of Pod termination in connection with the corresponding endpoint state and removal.
+
+This tutorial uses a simple nginx web server to demonstrate the concept.
+
+<!-- body -->
+
+#### Example flow with endpoint termination
+
+The following is the example of the flow described in the [Termination of Pods](/concepts/workloads/pods/pod-lifecycle/#pod-termination) document.
+
+Let's say you have a Deployment containing of a single `nginx` replica
+(just for demonstration purposes) and a Service:
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+  labels:
+    app: nginx
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: nginx
+  template:
+    metadata:
+      labels:
+        app: nginx
+    spec:
+      terminationGracePeriodSeconds: 120
+      containers:
+      - name: nginx
+        image: nginx:latest
+        ports:
+        - containerPort: 80
+        lifecycle:
+          preStop:
+            exec:
+              # just hang around for the duration of terminationGracePeriodSeconds
+              command: [
+                "/bin/sh", "-c", "sleep 120"
+              ]
+
+---
+
+apiVersion: v1
+kind: Service
+metadata:
+  name: nginx-service
+spec:
+  selector:
+    app: nginx
+  ports:
+    - protocol: TCP
+      port: 80
+      targetPort: 80
+```
+
+Once the Pod and Service are running, you can get the name of EndpointSlice:
+
+```shell
+kubectl get endpointslice
+```
+
+The output is similar to this:
+
+```shell
+NAME                  ADDRESSTYPE   PORTS   ENDPOINTS                 AGE
+nginx-service-6tjbr   IPv4          80      10.12.1.199,10.12.1.201   22m
+```
+
+You can see it's status and validate that there is one Endpoint registered:
+
+```shell
+kubectl get endpointslices -o json -l kubernetes.io/service-name=nginx-service
+```
+
+The output is similar to this:
+
+```json
+{
+    "addressType": "IPv4",
+    "apiVersion": "discovery.k8s.io/v1",
+    "endpoints": [
+        {
+            "addresses": [
+                "10.12.1.201"
+            ],
+            "conditions": {
+                "ready": true,
+                "serving": true,
+                "terminating": false
```

Comment:
This snippet will be syntax highlighted as a JSON object.
Since the object is incomplete, e.g. no ending brackets etc, the syntax parsing may fail.

If we only want to show an incomplete JSON object, we can replace `json` with `none` on line 94.

Reply:
most syntax highlighter ok with partial json. But happy to change, I understand the desire to make it as portable as possible.